cmd/crio/main.go

```diff
@@ -269,6 +269,18 @@ func main() {
 			return err
 		}
 
+		// Print the current cli flags
+		flagNames := c.FlagNames()
+		logrus.Infof("Cli Flags used:")
+		for _, flagName := range flagNames {
+			flagvalues := c.Value(flagName)
+			logrus.Infof("Flag: %v: %v \n", flagName, flagvalues)
+		}
```

Comment:
You can simplify this a little bit:

```suggestion
		for _, flagName := range c.FlagNames() {
			logrus.Infof("Flag: %s: %v\n", flagName, c.Value(flagName))
		}
```

However, this required a little bit more work.

Reply:
Take a look at the comment above:

- https://github.com/cri-o/cri-o/pull/7783/files#r1595020025

Then, have a look at the following:

```go
		// Print the current CLI flags.
		for _, flagName := range c.FlagNames() {
			flagValue := c.Value(flagName)
			// Turn a multi-value flag into a single command-separated list
			// of arguments, then wrap into a slice so that %v does it work
			// for us when rendering a slice type in the output.
			if _, ok := flagValue.(cli.StringSlice); ok {
				flagValue = []string{strings.Join(c.StringSlice(flagName), ",")}
			}
			logrus.Infof("Flag: %s: %v\n", flagName, flagValue)
		}
```
This might not be the cleanest way of doing this, but with this specific flag parsing package, there aren't a whole lot of things we could do here. 

The result is as expected per:

```console
$ crio --enable-criu-support --metrics-collectors=test1 --metrics-collectors=test2
(...)
INFO[2024-05-09 06:16:57.114618031Z] Cli Flags used:
INFO[2024-05-09 05:36:37.326853066Z] Flag: enable-criu-support: true
INFO[2024-05-09 05:36:37.326883781Z] Flag: metrics-collectors: [test1,test2]
```